src/lib.rs

```diff
@@ -0,0 +1,135 @@
+use crate::error::Error;
+use crate::html::{paper_wallets, to_data_url, WalletData};
+use bitcoin::{self, secp256k1, Address, AddressType, Network, PublicKey};
+use log::debug;
+use miniscript::bitcoin::secp256k1::SecretKey;
+use miniscript::bitcoin::PrivateKey;
+use miniscript::{self, Descriptor, DescriptorTrait, TranslatePk};
+use rand::{thread_rng, Rng};
+use std::collections::HashMap;
+
+mod error;
+mod html;
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug)]
+pub struct WifAndHexPub {
+    wif: String,
+    hex_pub: String,
+}
+
+/// Process descriptor and network and returns an html page encoded in a data-url which is pasteable
+/// in a browser
+pub fn process(descriptor: String, network: Network) -> Result<String> {
+    let descriptor_string: Descriptor<String> = descriptor.parse()?;
+    debug!("descriptor_string: {}", descriptor_string);
+
+    let (keys, address) = create_key_pairs_and_address(&descriptor_string, network)?;
+
+    let wallet_data = create_wallet_data(address, keys, &descriptor_string)?;
+
+    let html = paper_wallets(&wallet_data)?;
+
+    Ok(to_data_url(&html, "text/html"))
+}
+
+/// Creates a key pair for every alias in the given descriptor,
+/// Creates another descriptor replacing alias with the relative public key, so that we can compute the address.
+pub fn create_key_pairs_and_address(
+    descriptor_string: &Descriptor<String>,
+    network: Network,
+) -> Result<(HashMap<String, WifAndHexPub>, Address)> {
+    let mut rng = thread_rng();
+    let secp = secp256k1::Secp256k1::signing_only();
+    let mut keys = HashMap::new();
+    let alias_to_key = |alias: &String| -> Result<PublicKey> {
+        let entropy: [u8; 32] = rng.gen();
```

Comment:
if you generate the entropy externally maybe you can remove the `rand` feature from `bitcoin`

Reply:
Removed rand dep and restored generation through bitcoin rand feature (I did add rand because initially I experimented with bdk)